From 4b631262cee7cb156eef34e0f9500ad6ed711ac6 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Tue, 9 Aug 2016 18:11:47 +0200
Subject: Set default job error policy to "retry-job", since it is less

 confusing and a better default on most machines.
 .
 Amend documentation accordingly.
Author: Didier Raboud <odyx@debian.org>
Origin: vendor
Last-Update: 2016-07-03

Patch-Name: cups-set-default-error-policy-retry-job.patch
---
 doc/help/man-cupsd.conf.html | 4 ++--
 man/cupsd.conf.man.in        | 4 ++--
 scheduler/conf.c             | 6 +++---
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/doc/help/man-cupsd.conf.html b/doc/help/man-cupsd.conf.html
index bfe8e3dbb..2cdcecbdc 100644
--- a/doc/help/man-cupsd.conf.html
+++ b/doc/help/man-cupsd.conf.html
@@ -99,11 +99,11 @@ The default value is "30".
 <dt><a name="ErrorPolicy"></a><b>ErrorPolicy abort-job</b>
 <dd style="margin-left: 5.0em">Specifies that a failed print job should be aborted (discarded) unless otherwise specified for the printer.
 <dt><b>ErrorPolicy retry-job</b>
-<dd style="margin-left: 5.0em">Specifies that a failed print job should be retried at a later time unless otherwise specified for the printer.
+<dd style="margin-left: 5.0em">Specifies that a failed print job should be retried at a later time unless otherwise specified for the printer. The 'retry-job' error policy is the default.
 <dt><b>ErrorPolicy retry-this-job</b>
 <dd style="margin-left: 5.0em">Specifies that a failed print job should be retried immediately unless otherwise specified for the printer.
 <dt><b>ErrorPolicy stop-printer</b>
-<dd style="margin-left: 5.0em">Specifies that a failed print job should stop the printer unless otherwise specified for the printer. The 'stop-printer' error policy is the default.
+<dd style="margin-left: 5.0em">Specifies that a failed print job should stop the printer unless otherwise specified for the printer.
 <dt><a name="FilterLimit"></a><b>FilterLimit </b><i>limit</i>
 <dd style="margin-left: 5.0em">Specifies the maximum cost of filters that are run concurrently, which can be used to minimize disk, memory, and CPU resource problems.
 A limit of 0 disables filter limiting.
diff --git a/man/cupsd.conf.man.in b/man/cupsd.conf.man.in
index 4cede48bd..92c97744c 100644
--- a/man/cupsd.conf.man.in
+++ b/man/cupsd.conf.man.in
@@ -148,13 +148,13 @@ The default value is "30".
 Specifies that a failed print job should be aborted (discarded) unless otherwise specified for the printer.
 .TP 5
 \fBErrorPolicy retry-job\fR
-Specifies that a failed print job should be retried at a later time unless otherwise specified for the printer.
+Specifies that a failed print job should be retried at a later time unless otherwise specified for the printer. The 'retry-job' error policy is the default.
 .TP 5
 \fBErrorPolicy retry-this-job\fR
 Specifies that a failed print job should be retried immediately unless otherwise specified for the printer.
 .TP 5
 \fBErrorPolicy stop-printer\fR
-Specifies that a failed print job should stop the printer unless otherwise specified for the printer. The 'stop-printer' error policy is the default.
+Specifies that a failed print job should stop the printer unless otherwise specified for the printer.
 .\"#FilterLimit
 .TP 5
 \fBFilterLimit \fIlimit\fR
diff --git a/scheduler/conf.c b/scheduler/conf.c
index 5338df13e..f3c77630d 100644
--- a/scheduler/conf.c
+++ b/scheduler/conf.c
@@ -789,7 +789,7 @@ cupsdReadConfiguration(void)
   cupsdSetString(&LPDConfigFile, CUPS_DEFAULT_LPD_CONFIG_FILE);
   cupsdSetString(&SMBConfigFile, CUPS_DEFAULT_SMB_CONFIG_FILE);
 
-  cupsdSetString(&ErrorPolicy, "stop-printer");
+  cupsdSetString(&ErrorPolicy, "retry-job");
 
   JobHistory          = DEFAULT_HISTORY;
   JobFiles            = DEFAULT_FILES;
@@ -1313,8 +1313,8 @@ cupsdReadConfiguration(void)
       strcmp(ErrorPolicy, "retry-job") &&
       strcmp(ErrorPolicy, "stop-printer"))
   {
-    cupsdLogMessage(CUPSD_LOG_ALERT, "Invalid ErrorPolicy \"%s\", resetting to \"stop-printer\".", ErrorPolicy);
-    cupsdSetString(&ErrorPolicy, "stop-printer");
+    cupsdLogMessage(CUPSD_LOG_ALERT, "Invalid ErrorPolicy \"%s\", resetting to \"retry-job\".", ErrorPolicy);
+    cupsdSetString(&ErrorPolicy, "retry-job");
   }
 
  /*
